client/my-sites/checkout/composite-checkout/lib/paypal-express-processor.ts

```diff
@@ -32,23 +31,25 @@ export default async function payPalProcessor(
 		includeGSuiteDetails,
 		responseCart,
 		siteId,
+		siteSlug,
 	} = transactionOptions;
 	recordTransactionBeginAnalytics( {
 		reduxDispatch,
 		paymentMethodId: 'paypal',
 	} );
 
-	const { protocol, hostname, port, pathname } = parseUrl( window.location.href, true );
-
-	const successUrl = resolveUrl( window.location.href, getThankYouUrl() );
-
-	const cancelUrl = formatUrl( {
-		protocol,
-		hostname,
-		port,
-		pathname,
-		query: createUserAndSiteBeforeTransaction ? { cart: 'no-user' } : {},
-	} );
+	const thankYouUrl = getThankYouUrl();
+	let currentUrl;
+	try {
+		currentUrl = window.location.href;
+	} catch ( error ) {
+		currentUrl = `https://wordpress.com/checkout/${ siteSlug }`;
+	}
+	const successUrl = thankYouUrl.startsWith( 'http' ) ? thankYouUrl : currentUrl + thankYouUrl;
```

Comment:
Concatenating `currentUrl` and `thankYouUrl` here seems likely to be fragile.

Reply:
Wow, yes. How did I not see that before? Thank you. I've improved the logic here and used the base url instead of the current url to concatenate.